src/fparser/common/readfortran.py

```diff
@@ -1355,17 +1417,18 @@ def get_source_item(self):
                         line = get_single_line()
                         continue
                 else:
-                    # first line, check for a label
-                    m = _LABEL_RE.match(line)
-                    if m:
-                        assert not label, repr(label)
-                        label = int(m.group('label'))
-                        line = line[m.end():]
-                    # check for a construct name
-                    m = _CONSTRUCT_NAME_RE.match(line)
-                    if m:
-                        name = m.group('name')
-                        line = line[m.end():].lstrip()
+                    # Extract label from line if there is one.
+                    new_label, line = extract_label(line)
+                    # There may already be a label so only update if a
+                    # new one is found.
+                    if new_label:
```

Comment:
The original code threw an error if there was already a label. How might we already have a label and is it correct to throw it away?

Reply:
My mistake. I thought it was working in the same way as the multi-statement line code but didn't check properly. I've simplified.

Reply:
Thanks.